newIDE/app/src/ObjectEditor/Editors/SpriteEditor/AnimationPreview.js

```diff
@@ -270,16 +263,15 @@ const AnimationPreview = ({
               <Text>
                 <Trans>FPS:</Trans>
               </Text>
-              <TextField
+              <SemiControlledTextField
+                commitOnBlur
                 margin="none"
-                value={fps}
-                onChange={(e, text) => {
-                  const fps = parseFloat(text);
+                value={fps.toString()}
+                onChange={text => {
+                  if (!text) return;
+                  const fps = Number.parseFloat(text);
                   if (fps > 0) {
-                    setFps(fps);
-                    const newTimeBetweenFrames = parseFloat(
-                      (1 / fps).toFixed(4)
-                    );
+                    const newTimeBetweenFrames = 1 / fps;
```

Comment:
do you need to keep the toFixed(4) here too in case the division gives infinite decimals?

Reply:
It happens but it also adds unnecessary trailing 0 if the value is an integer so I prefer like this. I could just truncate the value after the dot though